packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -565,7 +593,19 @@ - (CVPixelBufferRef)copyPixelBuffer {
     }
   }
 
-  return buffer;
+  // Calling textureFrameAvailable only from within displayLinkFired would require a non-trivial
+  // solution to minimize missed video frames due to race between displayLinkFired, copyPixelBuffer
+  // and place where is _textureFrameAvailable reset to false in the flutter engine. Ideally
+  // FlutterTexture would support mode of operation where the copyPixelBuffer is called always or
+  // some other alternative, instead of on demand by calling textureFrameAvailable.
+  if (self.displayLink.running) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [self.frameUpdater.registry textureFrameAvailable:self.frameUpdater.textureId];
+    });
+  }
+
+  // Better to avoid returning NULL as it is unspecified what should be displayed in such a case.
+  return CVBufferRetain(self.latestPixelBuffer);
```

Comment:
And then here: ```// Add a retain for the engine, since the `copyPixelBufferForItemTime:` has already been accounted for, and the engine expects an owning reference.```